test/test_ssltransport.py

```diff
@@ -367,13 +367,10 @@ def test_wrong_sni_hint(self):
         with self.client_context.wrap_socket(
             sock, server_hostname="localhost"
         ) as proxy_sock:
-            with pytest.raises(Exception) as e:
+            with pytest.raises(ssl.SSLError):
```

Comment:
Can we catch https://docs.python.org/3/library/ssl.html#ssl.SSLCertVerificationError instead?